### ISSUE TITLE
LaunchDarklyReactNativeClient fails to compile, `React/RCTBridge.h file not found`

### ISSUE THREAD

**Issue by lfalkner** [launchdarkly/react-native-client-sdk#14]
**Is this a support request?**
No, it is specific to this SDK

**Describe the bug**
After following the integration guide, iOS still fails to compile. There is a warning when XCode starts the task, `BuildtargetLaunchDarklyReactNativeClient`, which says that it is missing the dependency target `React`.
Shortly thereafter, the step fails and errors are logged that the `React/RCTBridge` header cannot be found. (see attached screenshot).
We have attempted manually linking the React dependency, as well as verifying that the header search values are as expected (per various similar issues) in both the targets for `LaunchDarklyReactNativeClient` as well as our own.

**To reproduce**
Update our Podfile [according to the SDK instructions](https://docs.launchdarkly.com/docs/react-native-sdk-reference#section-getting-started)
Clean `Pods/` directory
Run `rm Podfile.lock && bundle exec pod install`
Clean derived data directory
Restart XCode
Clean project and build.

**Expected behavior**
Build passes

**Logs**
See attached screenshot

**SDK version**
2.0.0

**Language version, developer tools**
XCode 10.2.1
React-Native 59.9
Cocoapods 1.5.3
Ruby 2.3.7p456
Swift 5.0

**OS/platform**
Mac OS Mojave 10.14.5
iOS 12.2
XCode 10.2.1

![Screen Shot 2019-07-12 at 12 11 31 PM](https://user-images.githubusercontent.com/31742732/61147105-95226b80-a4a9-11e9-90a0-4aceaed4ffaa.png)

**Comment by torchhound** [comment]
I'll try a reproduce this, in the meantime would you please update your CocoaPods version and try again?

**Comment by lfalkner** [comment]
@torchhound is there a specific version of CocoaPods I should try? Latest?

**Comment by lfalkner** [comment]
I've updated to the latest stable CocoaPods version (1.7.4), and retried the same steps. I'm now seeing a different error, `No Such Module: Launch Darkly`
I am able to successfully build and deploy the hello-react-native app.

**Comment by bwoskow-ld** [comment]
Hi @lfalkner - I'm a colleague of @torchhound 's here at LaunchDarkly.

I couldn't help but notice the space in your module name: `Launch Darkly`. Can you update your module import to be for `LaunchDarkly` (without the space) and rebuild?

**Comment by lfalkner** [comment]
apologies @bwoskow-ld that was a typo in what I'd written here. There is no space in either the error message or the module import. Thanks though!

**Comment by bwoskow-ld** [comment]
Do you mind sharing the contents of your podfile?

**Comment by lfalkner** [comment]
Sure, I ended up cutting a second branch and going through the same steps as I'd done before, this time trying to focus on the use_frameworks! requirement (as we have firebase and there's some manual linking to be done). The content around LaunchDarkly remains the same. Thanks

```# Uncomment the next line to define a global platform for your project
platform :ios, '10.0'

require_relative '../node_modules/react-native-unimodules/cocoapods.rb'

# Uncomment the next line if you're using Swift or would like to use dynamic frameworks
use_frameworks!

target 'qbmobile' do

  # Pods for Firebase
  pod 'Firebase'
  pod 'Firebase/Core'
  pod 'Fabric', '~> 1.10.0'
  pod 'Crashlytics', '~> 3.13.1'
  pod 'RNFirebase', :path => '../node_modules/react-native-firebase/ios'

  rn_path = '../node_modules/react-native'
  rn_darkly_path = '../node_modules/launchdarkly-react-native-client-sdk/ios/'

  pod 'React', :path => rn_path, :subspecs => [
    'Core',
    'CxxBridge',
    'DevSupport',
    'RCTActionSheet',
    'RCTAnimation',
    'RCTGeolocation',
    'RCTImage',
    'RCTLinkingIOS',
    'RCTNetwork',
    'RCTSettings',
    'RCTText',
    'RCTVibration',
    'RCTWebSocket',
  ]

  pod 'yoga', :path => "#{rn_path}/ReactCommon/yoga/yoga.podspec"
  pod 'DoubleConversion', :podspec => "#{rn_path}/third-party-podspecs/DoubleConversion.podspec"
  pod 'glog', :podspec => "#{rn_path}/third-party-podspecs/glog.podspec"
  pod 'Folly', :podspec => "#{rn_path}/third-party-podspecs/Folly.podspec"

  # expo unimodules
  use_unimodules!


  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'

  pod 'react-native-webview', :path => '../node_modules/react-native-webview'

  pod 'react-native-netinfo', :path => '../node_modules/@react-native-community/netinfo'

  pod 'RNLanguages', :path => '../node_modules/react-native-languages'


  #launchdarkly-react-native-client-sdk dependencies

  pod 'LaunchDarkly', '4.1.0'

  pod 'LaunchdarklyReactNativeClient', path: rn_darkly_path


  post_install do |installer|

    # clean up malformed headers caused by dynamic frameworks setup
    rnfirebase = installer.pods_project.targets.find { |target| target.name == 'RNFirebase' }
    rnfirebase.build_configurations.each do |config|
      config.build_settings['HEADER_SEARCH_PATHS'] = '$(inherited) ${PODS_ROOT}/Headers/Public/**'
    end

    installer.pods_project.targets.each do |target|

      # The following is needed to ensure the "archive" step works in XCode.
      # It removes React & Yoga from the Pods project, as it is already included in the main project.
      # Without this, you'd see errors when you archive like:
      # "Multiple commands produce ... libReact.a"
      # "Multiple commands produce ... libyoga.a"

      targets_to_ignore = %w(React yoga)
      
      if targets_to_ignore.include? target.name
        target.remove_from_project
      end

    end
  end

  target 'qbmobileTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

# system("mkdir -p Pods/Headers/Public/FirebaseCore && cp Pods/FirebaseCore/Firebase/Core/Public/* Pods/Headers/Public/FirebaseCore/")```

**Comment by bwoskow-ld** [comment]
Thanks for sharing. I have a couple thoughts:

1. We have an [open feature request](https://github.com/launchdarkly/react-native-client-sdk/issues/13) for making the SDK work without `use_frameworks!`. Similar to your use case, that developer submitted their request because they use firebase.
2. I see that you're using Expo, however the LaunchDarkly React Native SDK isn't currently compatible with Expo because the SDK uses native modules.

**Comment by lfalkner** [comment]
Thanks for taking a look! 

1) It seems as though firebase has provided a work around for this in their latest stable version (5.5.5). As of right now, I'm able to compile their libraries with dynamic frameworks, as long as I use the scripts which copy their header files to the correct location.
2) Given that we'd not written our app in expo, and had added unimodules as a dependency after the fact, I'd understood that the compatibility issues between expo and native modules did not apply.

It seems as though the LaunchDarkly client does not correctly link to React as a dependency in our project - is there perhaps a configuration in our project which is missing?

**Comment by bwoskow-ld** [comment]
From looking over your podfile I don't see any missing configurations to link to React.

**Comment by dkwl93** [comment]
I've found that this doesn't resolve `pod 'LaunchDarkly', '4.1.0'` so I have specified where to resolve the dependency from manually. 

`pod 'LaunchDarkly', :git => 'https://github.com/launchdarkly/ios-client-sdk.git', :tag => '4.1.0'`

Please give that a go, I think it will resolve your issue of `No Such Module: LaunchDarkly`

CC: @lfalkner

**Comment by lfalkner** [comment]
@dkwl93 Thanks so much for the suggestion. When I gave that a try, I ran into the same issue I'd originally seen on the post, `React/RCTBridge.h file not found`.
Following more advice I'd seen online about this error, I tried manually adding React lib as a linked binary in the sdk's build phases tab in XCode. 
![Screen Shot 2019-07-18 at 11 42 57 AM](https://user-images.githubusercontent.com/31742732/61471780-567d2d00-a951-11e9-8fcc-70b1539178f8.png)

I'm now receiving an error that XCode cannot precompile the bridging header.
![Screen Shot 2019-07-18 at 11 44 21 AM](https://user-images.githubusercontent.com/31742732/61471838-744a9200-a951-11e9-948b-8cd8f9459973.png)
